### PR TITLE
Create new batch export/import page to/from ZIP 

### LINKED ISSUE #783: Batch Import (from ZIP)
From release plan spreadsheet

Estimate from release plan: 14 hours
Expressing interest: none

No additional notes

### PR DESCRIPTION
**Pull request by atarix83**
## References
_Add references/links to any related issues or PRs. These may include:_
* Fixes #783
* Requires DSpace/DSpace#8318

## Description
This PR add a new page to perform a new batch import via zip file

## Instructions for Reviewers

List of changes in this PR:
* Add new menu link for Batch Import Zip under import menu.
* Add a new page for the page import
1360937563-Screenshot 2022-09-13 at 6 25 49 PM

1889243600-Screenshot 2022-09-13 at 6 26 57 PM

3990256370-Screenshot 2022-09-13 at 6 28 11 PM

713209836-Screenshot 2022-09-13 at 6 30 53 PM

**Include guidance for how to test or review your PR.** This may include: steps to reproduce a bug, screenshots or description of a new feature, or reasons behind specific changes. 

## Checklist
- [x] My PR is small in size (e.g. less than 1,000 lines of code, not including comments & specs/tests), or I have provided reasons as to why that's not possible.
- [x] My PR passes [TSLint](https://palantir.github.io/tslint/) validation using `yarn run lint`
- [x] My PR doesn't introduce circular dependencies
- [x] My PR includes [TypeDoc](https://typedoc.org/) comments for _all new (or modified) public methods and classes_. It also includes TypeDoc for large or complex private methods.
- [x] My PR passes all specs/tests and includes new/updated specs or tests based on the [Code Testing Guide](https://wiki.lyrasis.org/display/DSPACE/Code+Testing+Guide).
- [x] If my PR includes new, third-party dependencies (in `package.json`), I've made sure their licenses align with the [DSpace BSD License](https://github.com/DSpace/DSpace/blob/main/LICENSE) based on the [Licensing of Contributions](https://wiki.lyrasis.org/display/DSPACE/Code+Contribution+Guidelines#CodeContributionGuidelines-LicensingofContributions) documentation.

### CODE REVIEW

File: src/app/core/data/processes/script-data.service.ts [hunk 1]
```diff
@@ -24,6 +24,7 @@ import { dataService } from '../base/data-service.decorator';
 
 export const METADATA_IMPORT_SCRIPT_NAME = 'metadata-import';
 export const METADATA_EXPORT_SCRIPT_NAME = 'metadata-export';
+export const BATCH_IMPORT_SCRIPT_NAME = 'batch-import';
```
Comment (tdonohue) [inline src/app/core/data/processes/script-data.service.ts:27]: This is not the correct name. The script is named `import` on the backend

File: src/app/shared/dso-selector/modal-wrappers/import-batch-selector/import-batch-selector.component.ts [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+ * Component to wrap a list of existing dso's inside a modal
+ * Used to choose a dso from to import metadata of
+ */
+@Component({
+  selector: 'ds-import-batch-selector',
+  templateUrl: '../dso-selector-modal-wrapper.component.html',
+})
+export class ImportBatchSelectorComponent extends DSOSelectorModalWrapperComponent implements OnInit {
+  objectType = DSpaceObjectType.DSPACEOBJECT;
+  selectorTypes = [DSpaceObjectType.COLLECTION, DSpaceObjectType.COMMUNITY];
```
Comment (tdonohue) [inline src/app/shared/dso-selector/modal-wrappers/import-batch-selector/import-batch-selector.component.ts:21]: This looks incorrect, as you can only import into a Collection, not a Community

File: src/assets/i18n/en.json5 [hunk 3]
```diff
@@ -542,28 +542,45 @@
...
 
+  "admin.batch-import.title": "Import Batch",
+
   "admin.metadata-import.page.header": "Import Metadata",
 
+  "admin.batch-import.page.header": "Import Batch",
+
   "admin.metadata-import.page.help": "You can drop or browse CSV files that contain batch metadata operations on files here",
 
+  "admin.batch-import.page.help": "You can drop or browse ZIP files that contain batch operations on files here",
```
Comment (tdonohue) [inline src/assets/i18n/en.json5:557]: I think we should clarify this description.  You cannot perform "batch operations"... this is just an import.

Perhaps change the description to say:
`Select the Collection to import into. Then, drop or browse to a Simple Archive Format (SAF) zip file which includes the Items you wish to import.`

**Review by tdonohue — CHANGES_REQUESTED** [review]
@atarix83 : This PR currently doesn't work because it has the wrong script name.  It's trying to run a `batch-import` script, but the backend PR defines the script as being named `import`.  I've noted inline where there appear to be bugs in this code.

Additionally, is there any way to update this PR to also show a `Batch Export (ZIP)` option in the menu?  The backend PR also defines a new `export` script which can be used to export the content of a Collection.  I expect it could look very similar to the existing "Export -> Metadata" menu, except that it should allow you to export from a Collection only (it is not supported for Communities).

**Review by tdonohue — CHANGES_REQUESTED** [review]
@atarix83 : Currently, the `export` does NOT work but the `import` *does* work.

* **BUG:** Selecting "Export -> Batch Export (ZIP)" correctly gives me a list of Collections to choose from.  However, once I select a Collection the process immediately fails.  It appears that it has an extra `-a` parameter which is invalid and causes the failure.  (Running the same command without `-a` works fine).

I've verified that the `import` works fine both when "Validate Only" is selected and deselected.  (NOTE for other testers, if you plan to import the same Zip you exported, make sure to export with the `--migrate` flag. That way the import process doesn't try to restore the handles.)

**Comment by atarix83** [comment]
@tdonohue 

> BUG: Selecting "Export -> Batch Export (ZIP)" correctly gives me a list of Collections to choose from. However, once I select a Collection the process immediately fails. It appears that it has an extra -a parameter which is invalid and causes the failure. (Running the same command without -a works fine).

should be fixed now

**Review by tdonohue — APPROVED** [review]
👍 Thanks @atarix83 and @nikunj59  !  Retested today and verified this fully works now & all my prior feedback has been addressed.

**Review by mspalti — APPROVED** [review]
@atarix83 , @tdonohue , I tested `import` and `export` and everything worked great from the UI side. I see a problem with the `--help` that I mention in DSpace/DSpace/pull/8318. Basically, the `--help` option is redundant in the Angular view and provides no useful output, but I see it listed. The problem is minor and needs to be fixed on the backend side so I can approve here.

**Pull request merged**